liquidprompt

```diff
@@ -150,6 +150,175 @@ case "$(uname)" in
+                ;;
+            Darwin)
+                if (( LP_ENABLE_OS_FAMILY )); then lp_os_family=BSD; fi
+                if (( LP_ENABLE_OS_KERNEL )); then lp_os_kernel=Darwin; fi
+                ;;
+            SunOS)
+                if (( LP_ENABLE_OS_FAMILY )); then lp_os_family=UNIX; fi
+                if (( LP_ENABLE_OS_KERNEL )); then lp_os_kernel=SunOS; fi
+                ;;
+            CYGWIN*)
+                if (( LP_ENABLE_OS_FAMILY )); then lp_os_family=Windows; fi
+                if (( LP_ENABLE_OS_KERNEL )); then lp_os_kernel=Cygwin; fi
+                ;;
+            MSYS*)
+                if (( LP_ENABLE_OS_FAMILY )); then lp_os_family=Windows; fi
+                if (( LP_ENABLE_OS_KERNEL )); then lp_os_kernel=MSYS; fi
+                ;;
+            MINGW*)
+                if (( LP_ENABLE_OS_FAMILY )); then lp_os_family=Windows; fi
+                if (( LP_ENABLE_OS_KERNEL )); then lp_os_kernel=MinGW; fi
+                ;;
+            *) # Assume Linux
+                if (( LP_ENABLE_OS_KERNEL )); then lp_os_kernel=Linux; fi
+                if (( LP_ENABLE_OS_FAMILY )); then
+                    if [[ "$kernel" == "Linux" ]]; then
+                        osfamily="$(uname -o)"
+                        if [[ "$osfamily" == "Android" ]]; then
+                            lp_os_family=Android
+                            return
+                        fi # Android
+                    fi # Linux
+                    lp_os_family=GNU
+                fi
+                # TODO parse HOME_URL and add an hyperlink.
+                if (( LP_ENABLE_OS_DISTRIB && LP_ENABLE_OS_VERSION )); then
+                    if _lp_grep_fields "/etc/os-release" "=" "ID" "VERSION_CODENAME"; then
+                        lp_os_distrib=${lp_grep_fields[_LP_FIRST_INDEX+0]-}
+                        lp_os_version=${lp_grep_fields[_LP_FIRST_INDEX+1]-}
+                    else
+                        return "$?"
+                    fi
+                elif (( LP_ENABLE_OS_DISTRIB )); then
+                    if _lp_grep_fields "/etc/os-release" "=" "ID"; then
+                        lp_os_distrib=${lp_grep_fields[_LP_FIRST_INDEX+0]-}
+                    else
+                        return "$?"
+                    fi
+                elif (( LP_ENABLE_OS_VERSION )); then
+                    if _lp_grep_fields "/etc/os-release" "=" "VERSION_CODENAME"; then
+                        lp_os_version=${lp_grep_fields[_LP_FIRST_INDEX+0]-}
+                    else
+                        return "$?"
+                    fi
+                fi
+                ;;
+        esac
+    fi
+}
+
+_lp_os_color() {
+    # Substitute lp_os_* names with icons/colored strings (from LP_MARK_OS),
+    # or hash colored versions.
+    # Join everything with LP_MARK_OS_SEP.
+    _lp_os || return "$?"
+
+    lp_os_color=
+    local active=()
+    if (( LP_ENABLE_OS_ARCH )); then
```

Comment:
Since these strings are not always going to be filled, why not just check if the string has contents?
```suggestion
    if [[ -n $lp_os_arch ]]; then
```
That removes a double check of the config option, as well as protecting against situations where the data was not found.

Reply:
Clever!